src/utils/account.ts

```diff
@@ -20,10 +20,18 @@ export async function loginWithSecretKey(secretKey: string): Promise<Signer> {
   const { Keyring } = await import('@polkadot/keyring')
   const keyring = new Keyring({ type: 'sr25519' })
   const secret = Buffer.from(secretKey, 'hex')
-  const pair = keyring.addFromSeed(secret, {}, 'ed25519')
-  return pair
+  const signer = keyring.addFromSeed(secret, {}, 'ed25519')
```

Comment:
```suggestion
  const signer = keyring.addFromSeed(secret, {}, 'sr25519')
```

Reply:
Because `sr25519` is the default in Polkadot wallets, and if you get a private key from Grill what was generated with `ed25519` and put it to Polkadot.js so we will generate a new key :)

Reply:
But sr25519 expects WASM, so check maybe need to call `waitReady` from `polkadot-wasm`

Reply:
Updated in [65ad947](https://github.com/dappforce/grillchat/pull/18/commits/65ad947d193657ef8bd53b49aa9db102a2a23812)

Reply:
Checked it works on private key logout/login

Reply:
also worked on share session